packages/itwin/measure-tools/src/measurements/PerpendicularDistanceMeasurement.ts

```diff
@@ -0,0 +1,285 @@
+        value: fDistance,
+        aggregatableValue: lengthSpec !== undefined ? { value: distance, formatSpec: lengthSpec } : undefined,
+      },
+      {
+        label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.run"),
+        name: "DistanceMeasurement_Run",
+        value: fRun,
+        aggregatableValue: lengthSpec !== undefined ? { value: run, formatSpec: lengthSpec } : undefined,
+      },
+      {
+        label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.rise"),
+        name: "DistanceMeasurement_Rise",
+        value: fRise,
+      },
+      {
+        label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.slope"),
+        name: "DistanceMeasurement_Slope",
+        value: fSlope,
+      },
+    );
+
+    if (this.drawingMetadata?.worldScale === undefined) {
+      data.properties.push(
+        {
+          label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.delta_x"),
+          name: "DistanceMeasurement_Dx",
+          value: fDeltaX,
+        },
+        {
+          label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.delta_y"),
+          name: "DistanceMeasurement_Dy",
+          value: fDeltaY,
+        },
+        {
+          label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.startCoordinates"),
+          name: "DistanceMeasurement_StartPoint",
+          value: fStartCoords,
+        },
+        {
+          label: MeasureTools.localization.getLocalizedString("MeasureTools:tools.MeasureDistance.endCoordinates"),
+          name: "DistanceMeasurement_EndPoint",
+          value: fEndCoords,
+        },
+      );
+    }
+
+    return data;
+  }
+
+  /**
+   * Tests equality with another measurement.
+   * @param other Measurement to test equality for.
+   * @param opts Options for equality testing.
+   * @returns true if the other measurement is equal, false if some property is not the same or if the measurement is not of the same type.
+   */
+  public override equals(other: Measurement, opts?: MeasurementEqualityOptions): boolean {
+    if (!super.equals(other, opts)) return false;
+
+    // Compare data (ignore isDynamic)
+    const tol = opts ? opts.tolerance : undefined;
+    const otherDist = other as PerpendicularDistanceMeasurement;
+    if (
+      otherDist === undefined ||
+      !this.startPointRef.isAlmostEqual(otherDist.startPointRef, tol) ||
+      !this.endPointRef.isAlmostEqual(otherDist.endPointRef, tol) ||
+      this.showAxes !== otherDist.showAxes
+    )
+      return false;
+
+    return true;
+  }
+
+  /**
+   * Copies data from the other measurement into this instance.
+   * @param other Measurement to copy property values from.
+   */
+  protected override copyFrom(other: Measurement) {
+    super.copyFrom(other);
+
+    if (other instanceof PerpendicularDistanceMeasurement) {
```

Comment:
Done

Reply:
@saaaaaally should do a deep copy of `other._secondaryLine`, otherwise callers can get into trouble changing the points causing a change in the original measurement.

Reply:
Done